pkg/scheduler/preemption/preemption.go

```diff
@@ -395,6 +403,28 @@ func workloadFits(wlReq cache.FlavorResourceQuantities, cq *cache.ClusterQueue,
 	return true
 }
 
+// is queue exhausted for all requested flavors
+func isQueueExhaustedForAllRequestedFlavors(wl *workload.Info, assignment flavorassigner.Assignment, cq *cache.ClusterQueue) bool {
+	wlReq := totalRequestsForAssignment(wl, assignment)
```

Comment:
This line is repeated with `minimalPreemptions`. I think we should move it up, but for now keep it here to make the diff minimal.

Reply:
Done